app/scripts/modules/core/authentication/authentication.initializer.service.ts

```diff
@@ -70,12 +71,23 @@ export class AuthenticationInitializer {
     this.redirectService.redirect(`${this.settings.gateUrl}/auth/redirect?to=${callback}`);
   }
 
+  public openUserRolesModal() {
+    this.$uibModal.open({
+      templateUrl: require('./userMenu/checkUserRolesModal.html'),
+      controller: 'CheckUserRolesModalCtrl as ctrl',
+      size: 'squared',
+      resolve: {
+        userRoles: () => this.authenticationService.getAuthenticatedUser().roles
+      }
+    });
+  }
+
   public authenticateUser() {
     this.$rootScope.authenticating = true;
     this.$http.get(this.settings.authEndpoint)
       .then((response: ng.IHttpPromiseCallbackArg<IAuthResponse>) => {
         if (response.data.username) {
-          this.authenticationService.setAuthenticatedUser(response.data.username);
+          this.authenticationService.setAuthenticatedUser(response.data.username, response.data.roles);
```

Comment:
I think a better solution here is to create a new `interface` type to pass in that encapsulates both pieces of data (since they are tightly-coupled) - such that the `roles` property is optional.